### ISSUE TITLE
[data grid] An option to show column group name next to the column name in the GridColumnsPanel

### ISSUE THREAD

**Issue by VladyslavBondarenko** [mui/mui-x#9388]
### Summary 💡

The columns panel opened on "Manage columns" from the column menu shows the columns list with their visibility toggles.

It would be useful to have the option to show the column group name next to the column name.

Or if getting a column group name is complicated (because you'd need to handle the case of nested column groups), would be nice to have the option to have something like `headerNameExtended` in the column props and have `label={column.headerNameExtended || column.headerName || column.field}` [here](https://github.com/mui/mui-x/blob/master/packages/grid/x-data-grid/src/components/panel/GridColumnsPanel.tsx#LL244C17-L244C17). Or any other solution.

### Motivation 🔦

We have a table with many column groups, where each group is some metric, and each group has two columns - "24 h" and "30 min". It's not the best UX when the columns panel has a list of many "12 h" and "30 min" and it's not immediately clear to which metric each column belongs.
I can think of some workaround, but it could also be useful for others.

I saw https://github.com/mui/mui-x/issues/5700, but I see it'll take to be implemented, wanted to know your opinion on whether the `headerNameExtended` prop or any other workaround could be added earlier.

### Order ID 💳 (optional)

64622

**Comment by VladyslavBondarenko** [comment]
The easiest workaround I found that doesn't require changes in the data-grid  is:

For group columns:
- add `renderHeader` that returns `headerName`
- add a group name to the `headerName`

This way, in the table header: `renderHeader` is called and only column name is rendered. In the columns panel: `headerName` is rendered which includes a group name.

It's working, but isn't elegant and isn't future-proof.

**Comment by romgrk** [comment]
We'll look into this, but we have a lot of issues to work on. If there is more interest (:+1:s) we may prioritize it more.

**Comment by mimaxm** [comment]
@VladyslavBondarenko
Hello! Tell me, have you found a solution to this problem?
I need to combine 2-3 columns into one group. The group names are always different, but the column names may be the same. I need to display some column names in the table header, and different names in the filters and column manager.

**Comment by VladyslavBondarenko** [comment]
Yes, see [my comment](https://github.com/mui/mui-x/issues/9388#issuecomment-1597063707). You can set `headerName` with what you want to see in column manager and set `renderHeader` with what you want to see in the table header.

<details>
<summary>Here is the exact solution if this is what you're interested in</summary>

```
const findColumnGroup = (columnField, columnGroupingModel) =>
  columnGroupingModel?.find(group =>
    group.children.some(groupColumn => groupColumn.field === columnField)
  );
  
...

columns.map(column => {
  const columnGroup = findColumnGroup(column.field, columnGroupingModel);

  if (columnGroup) {
    const headerName = column.headerName ?? column.field;

    return {
      ...column,
      headerName: `${columnGroup.groupId}: ${headerName}`,
      renderHeader: () => column.renderHeader?.() ?? headerName
    };
  }

  return column;
});
```

</details>
